### DISCUSSION TITLE
Make `Vec3A::from_vec4` a `const fn`

### DISCUSSION THREAD

**Discussion by herabit** [bitshifter/glam-rs#546] [Ideas]
All of the implementations of `Vec3A::from_vec4` do not seem to require non-const functionality, so marking them all as `const fn` will increase their versatility for API consumers.

**Comment by bitshifter** [comment]
Hi, yep that sounds good, thanks!

In case you aren't familiar with glam internals, you need to make the change in `codegen/templates/vec.rs.tera` and then `cargo run -p codegen` to regenerate the vec4 sources.

**Comment by bitshifter** [comment]
Ah it turns out this is not currently possible because Vec3::from_vec4 invokes deref to access the x,y & z values which can't be done in a const context.
